File: medium/max-increase-to-keep-city-skyline.swift
class Solution {
    func maxIncreaseKeepingSkyline(_ grid: [[Int]]) -> Int {
        var rowMax = [Int]()
        var colMax = [Int]()
        
        // row max
        for row in grid {
            rowMax.append(row.max()!)
        }
        
        // col max
        for colIndex in 0..<grid[0].count {
            var cm = 0
            for rowIndex in 0..<grid.count {
                cm = max(cm, grid[rowIndex][colIndex])
            }
            colMax.append(cm)
        }

        var answer = 0
        for row in 0..<grid.count {
            for col in 0..<grid[0].count {
                answer += (min(rowMax[row], colMax[col]) - grid[row][col])
            }
        }
        return answer
    }
}

File: medium/max-increase-to-keep-city-skyline.py
class Solution:
    def maxIncreaseKeepingSkyline(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        
        transgrid = []
        transgrid = copy.deepcopy(grid)

        tb = []
        lr = []

        tbmax = 0
        lrmax = 0

        for i in range(len(grid)):
            for j in range(len(grid[0])):
                transgrid[j][i] = grid[i][j]

        for l in grid:
            lr.append(max(l))
        for l in transgrid:
            tb.append(max(l))

        for i in range(len(grid)):
            for j in range(len(grid[0])):
                grid[i][j] = min(lr[i], tb[j])

        sum = 0
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                sum += (grid[i][j] - transgrid[i][j])



        return sum

